libs/@local/hql/diagnostics/src/config.rs

```diff
@@ -0,0 +1,69 @@
+use ariadne::{CharSet, IndexType, LabelAttach};
+
+/// Mirror of [`ariadne::Config`]
+///
+/// This is needed for access to the `ariadne::Config` struct
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[expect(clippy::struct_excessive_bools, reason = "mirror of ariadne::Config")]
+pub struct ReportConfig<F> {
+    pub cross_gap: bool,
+    pub label_attach: LabelAttach,
+    pub compact: bool,
+    pub underlines: bool,
+    pub multiline_arrows: bool,
+    pub color: bool,
+    pub tab_width: usize,
+    pub char_set: CharSet,
+    pub index_type: IndexType,
+    pub transform_span: F,
+}
+
+impl<F> ReportConfig<F> {
+    // just makes creating it easier
```

Comment:
That would make `ReportConfig::default()` not work in conjunction with `with_transform_span`, and you can't do:

```
ReportConfig {
	transform_span: |span| DiagnosticSpan { .. },
	..ReportConfig::default()
}
```

because fn doesn't implement `transform_span`

Reply:
fair enough!